### PR TITLE
feat (docs): document data stream protocol for useChat and useCompletion

### PR DESCRIPTION
**Pull request by lgrammel**

### CODE REVIEW

File: content/docs/05-ai-sdk-ui/50-stream-protocol.mdx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+---
+
+# Stream Protocols
+
+AI SDK UI functions such as `useChat` and `useCompletion` support both text streams and data streams.
+The stream protocol defines how the data is streamed to the frontend on top of the HTTP protocol.
+
+This page describes both protocols and how to use them in the backend and frontend.
+
+You can use this information develop custom backends and frontends for your use case, e.g.,
```
Comment (nicoalbanese) [inline content/docs/05-ai-sdk-ui/50-stream-protocol.mdx:13]: "to develop"

Comment (jeremyphilemon) [inline content/docs/05-ai-sdk-ui/50-stream-protocol.mdx:13]: "this information (to) develop"

File: content/docs/05-ai-sdk-ui/50-stream-protocol.mdx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  const result = await streamText({
+    model: openai('gpt-4o'),
+    prompt,
+  });
+
+  return result.toTextStreamResponse();
+}
+```
+
+## Data Streams
```
Comment (jeremyphilemon) [inline content/docs/05-ai-sdk-ui/50-stream-protocol.mdx:77]: "Data Stream Protocol" to match above "Text Stream Protocol"

File: content/docs/05-ai-sdk-ui/50-stream-protocol.mdx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+The stream protocol defines how the data is streamed to the frontend on top of the HTTP protocol.
+
+This page describes both protocols and how to use them in the backend and frontend.
+
+You can use this information develop custom backends and frontends for your use case, e.g.,
+to provide compatible API endpoints that are implemented in a different language such as Python.
+
+## Text Stream Protocol
+
+Text streams are simple text chunks that are streamed to the frontend.
```
Comment (jeremyphilemon) [inline content/docs/05-ai-sdk-ui/50-stream-protocol.mdx:18]: "A text stream contains chunks in plain text, that are streamed to the frontend. Each chunk is then appended together to form a full text response."

File: content/docs/05-ai-sdk-ui/50-stream-protocol.mdx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    prompt,
+  });
+
+  return result.toTextStreamResponse();
+}
+```
+
+## Data Streams
+
+Data streams are a special protocol that the Vercel AI SDK provides to stream data to the frontend.
```
Comment (jeremyphilemon) [inline content/docs/05-ai-sdk-ui/50-stream-protocol.mdx:79]: "A data stream is a special protocol that..."

**Review by jeremyphilemon — APPROVED** [review]

**Pull request merged**